libvast/src/option_declaration_set.cpp

```diff
@@ -291,13 +297,34 @@ option_declaration_set::parse(option_map& xs, argument_iterator begin,
     else if (detail::starts_with(*first, "-"))
       return parse_short_option(first, last);
     else
-      return std::make_pair(parse_state::begin_is_not_an_option, first);
+      return std::make_pair(parse_state::not_an_option, first);
   };
   auto [state, it] = dispatch(begin, end);
   while (state == parse_state::in_progress)
     std::tie(state, it) = dispatch(it, end);
   return std::make_pair(state, it);
 }
 
+namespace {
+
+constexpr const char* parser_state_names[] = {
+  "successful",
+  "option already exists",
+  "not an option",
+  "name not declared",
+  "argument passed but not declared",
+  "argument declared but not passed",
+  "failed to parse argument",
+  "type not parsebale",
+  "in progress"
+};
+
+} // namespace anonymous
+
+const char* to_string(option_declaration_set::parse_state x) {
+  if (x >= option_declaration_set::parse_state::last_state)
```

Comment:
`parse_state` is an enum value that we never receive from external user input, right? In that case I would put an assert here instead of performing a bound check on every lookup.